metalware-new/src/metal_log.rb

```diff
@@ -50,17 +48,31 @@ def metal_log
     end
 
     def initialize(log_name)
-      file = "#{self.class.config.log_path}/#{log_name}.log"
+      file = "#{config.log_path}/#{log_name}.log"
       FileUtils.mkdir_p File.dirname(file)
       f = File.open(file, "a")
       f.sync = true
       super(f)
-      level = self.class.config.log_serverity
+      self.level = config.log_severity
     end
 
     def warn(msg)
-      Output.stderr "warning: #{msg}"
-      super msg
+      if config.cli.strict
+        e = StrictWarningError.new(msg)
+        e.set_backtrace(caller(2))
```

Comment:
Up until this point we don't have an error. So I needed to create the error object. However I am setting the backtrace to be where the warning was raised from and not from inside `MetalLog`.